src/query.c

```diff
@@ -1080,7 +1079,10 @@ static IndexIterator *Query_EvalTagPrefixNode(QueryEvalCtx *q, TagIndex *idx, Qu
   } else {    // TAG field has suffix triemap
     arrayof(char**) arr = GetList_SuffixTrieMap(idx->suffix, tok->str, tok->len,
                                                 qn->pfx.prefix, q->sctx->timeout);
-    if (!arr) return NULL;
+    if (!arr) {
+      rm_free(its);
+      return NULL;
+    }
```

Comment:
Check out - there is another `if (!arr) return NULL;` on line 1047.
I think the better fix would be setting `its` to `NULL` at the declaration, instead of using `rm_calloc` with capacity 0

Reply:
Nice catch!
The initial capacity isn't 0, it's 8.